packages/core/src/v3/resource-catalog/index.ts

```
@@ -0,0 +1,79 @@
+import { type ResourceCatalog } from "./catalog.js";
+import { NoopResourceCatalog } from "./noopResourceCatalog.js";
+
+const NOOP_RESOURCE_CATALOG = new NoopResourceCatalog();
+
+export class ResourceCatalogAPI {
+  private static _instance?: ResourceCatalogAPI;
+
+  private constructor() {}
+
+  public static getInstance(): ResourceCatalogAPI {
+    if (!this._instance) {
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)